src/radical/pilot/pmgr/launching/base.py

```diff
@@ -881,6 +881,7 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
         agent_cfg['task_post_exec']      = task_post_exec
         agent_cfg['resource_cfg']        = copy.deepcopy(rcfg)
         agent_cfg['debug']               = self._log.getEffectiveLevel()
+        agent_cfg['services']            = services
```

Comment:
please check this file again.  `services` is defined in two different places and is used in two different places.

Reply:
I check but did not find any specific problem. Can you help more here

Reply:
Check the occurence of `services` in that file - there should be 4 matching lines.  Do you need all of them?  I think you need 2.

Reply:
still didn't get that properly. 
Do you think I would not need this `agent_cfg['services']            = services` ?

Reply:
Ok, let's do step by step:

> Check the occurence of services in that file

Can you paste the lines you find?

Reply:
`SERVICES          = 'services'`

`SERVICES        : [TaskDescription],`

`        SERVICES        : []         ,`

I don't see a problem here, I am sure I am missing something very basic here. 
As per my understanding `SERVICES` var is defined in only 1 place, and used in 2 places

Reply:
I think you are looking in the wrong file - this looks different for me:

```sh
$ git pull
Already up to date.

$ grep services src/radical/pilot/pmgr/launching/base.py 
        services         = pilot['description']['services']
        services               += rcfg.get('services', [])
        for arg in services       :   bs_args.extend(['-j', arg])
        agent_cfg['services']            = services
```

Services are pulled from the pilot description, which is correct, but are also still pulled from the resource config which I would think we don't need anymore?  But, more importantly, they are passed via the resource config, which is ok, but still also passed as bootstrapper arguments, which does not make sense at all.

Reply:
ok now it get. I have removed the argument -j thing, but kept the resource config way and description.